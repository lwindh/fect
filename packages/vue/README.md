<p align="center" height="170">
  <img  style="height:170px;width:170px;" height="170" width="170" src="https://user-images.githubusercontent.com/52351095/118687359-7e809480-b837-11eb-8083-b0504ec79652.png"/>
</p>

<p align="center">
 <img src="https://img.shields.io/npm/dm/@fect-ui/vue?style=for-the-badge" alt="downloads" />
 <img src="https://img.shields.io/npm/v/@fect-ui/vue?color=%230761d1&logoColor=%23000000&style=for-the-badge" alt="npm Version" />
 <img src="https://img.shields.io/codecov/c/gh/fect-org/fect?style=for-the-badge" alt="Coverage Status" />
 <img src="https://img.shields.io/github/workflow/status/fect-org/fect/CI?style=for-the-badge" alt="CI Status" />
</p>

> Modern and minimalist Vue3 UI library,originating from Geist

## Quick start

1. run `yarn add @fect-ui/vue` or `npm install @fect-ui/vue` install it

2. import into project

```js
import { createApp } from 'vue'
import App from './App.vue'
import FectUI from '@fect-ui/vue'
import '@fect-ui/vue/lib/main.css'

createApp(App).use(FectUI).mount('#app')
```

## Browser Support

Fect supports modern browsers and Chrome >= 51、iOS >= 10.0 (same as Vue 3).

## Documentation

- [Canary Document](https://vue-rc.miaya.art/en-us)

## All Contributors

Thanks to the following friends for their contributions to Fect:

<a href="https://github.com/fect-org/fect/graphs/contributors">
  <img src="https://opencollective.com/fect/contributors.svg?width=890&button=false" alt="contributors">
</a>

## Development

- [Contributing Guide](https://github.com/fect-org/fect/blob/master/.github/CONTRIBUTING.md)

## Acknowledgements

Thanks to [JetBrains](https://www.jetbrains.com/) for allocating free open-source licences for IDEs such as WebStorm.

<p align="right">
<img width="250px" height="250px" src="https://resources.jetbrains.com/storage/products/company/brand/logos/jb_square.png" alt="JetBrains Black Box Logo logo.">
</p>

## LICENSE

[MIT](./LICENSE)
