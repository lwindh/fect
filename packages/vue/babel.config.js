module.exports = {
  presets: ['@fect-ui/cli/lib/config/babel-preset']
}
