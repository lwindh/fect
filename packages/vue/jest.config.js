module.exports = {
  testPathIgnorePatterns: ['/node_modules/', '/dist/', '/lib/', '/es/'],
  collectCoverageFrom: [
    'components/**/*.{js,jsx,ts,tsx,vue}',
    '!components/utils/**',
    '!components/index.ts',
    '!components/**/type.ts',
    '!components/**/style.ts',
    '!components/back-top/**'
  ],
  collectCoverage: true,
  preset: '@fect-ui/cli/lib/config/@jest',
  testTimeout: 20000
}
