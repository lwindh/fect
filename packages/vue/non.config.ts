import path from 'path'
import { defineNonConfig } from '@fect-ui/cli'

export default defineNonConfig({
  entry: path.join(__dirname, 'components'),
  lib: {
    input: path.join(__dirname, 'components'),
    format: 'default',
    name: 'fect'
  }
})
