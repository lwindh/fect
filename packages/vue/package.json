{
  "name": "@fect-ui/vue",
  "version": "1.2.0-rc.0",
  "description": "Minimalist UI components built on Vue",
  "main": "lib/index.js",
  "module": "es/index.js",
  "types": "types/index.d.ts",
  "jsdelivr": "lib/fect.min.umd.js",
  "scripts": {
    "collect": "non collect",
    "build": "npm run collect && non compile",
    "test": "non test",
    "volar": "non volar"
  },
  "keywords": [
    "vue component",
    "fect",
    "vue ui"
  ],
  "homepage": "https://github.com/fect-org/fect",
  "repository": {
    "type": "git",
    "url": "https://github.com/fect-org/fect"
  },
  "files": [
    "lib/",
    "es/",
    "types/",
    "vetur/",
    "README.md",
    "components.d.ts"
  ],
  "author": "XeryYue",
  "license": "MIT",
  "devDependencies": {
    "@fect-ui/cli": "^1.0.0"
  },
  "sideEffects": [
    "es/**/style/*",
    "lib/**/style/*",
    "*.css"
  ],
  "dependencies": {
    "@fect-ui/themes": "^2.0.0",
    "@fect-ui/vue-hooks": "^1.0.0",
    "proy": "^0.0.3"
  },
  "peerDependencies": {
    "vue": "^3.0.11"
  }
}
