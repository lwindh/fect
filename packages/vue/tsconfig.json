{
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "/",
    "jsx": "preserve",
    "target": "esnext",
    "module": "esnext",
    "strict": true,
    "allowJs": true,
    "noImplicitThis": true,
    "importHelpers": true,
    "esModuleInterop": true,
    "moduleResolution": "node",
    "types": ["jest"],
    "lib": ["esnext", "dom", "dom.iterable", "scripthost"]
  },
  "include": ["components", "../../shims/*"],
  "exclude": ["node_modules"]
}
