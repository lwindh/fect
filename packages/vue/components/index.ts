const version = '1.2.0-rc.1'

import { App } from 'vue'

import { Avatar } from './avatar'
import { AvatarGroup } from './avatar-group'
import { BackTop } from './back-top'
import { Badge } from './badge'
import { BadgeAnchor } from './badge-anchor'
import { Breadcrumbs } from './breadcrumbs'
import { BreadcrumbsItem } from './breadcrumbs-item'
import { Button } from './button'
import { Capacity } from './capacity'
import { Card } from './card'
import { Checkbox } from './checkbox'
import { CheckboxGroup } from './checkbox-group'
import { Code } from './code'
import { Col } from './col'
import { Collapse } from './collapse'
import { CollapseGroup } from './collapse-group'
import { Dot } from './dot'
import { Drawer } from './drawer'
import { Form } from './form'
import { FormItem } from './form-item'
import { Grid } from './grid'
import { GridGroup } from './grid-group'
import { Image } from './image'
import { ImageBrowser } from './image-browser'
import { Input } from './input'
import { Link } from './link'
import { Loading } from './loading'
import { Modal } from './modal'
import { Pagination } from './pagination'
import { Popover } from './popover'
import { Progress } from './progress'
import { Radio } from './radio'
import { RadioGroup } from './radio-group'
import { Rating } from './rating'
import { Row } from './row'
import { Select } from './select'
import { SelectOption } from './select-option'
import { Skeleton } from './skeleton'
import { SkeletonItem } from './skeleton-item'
import { Slider } from './slider'
import { Snippet } from './snippet'
import { Spacer } from './spacer'
import { Spinner } from './spinner'
import { Swipe } from './swipe'
import { SwipeItem } from './swipe-item'
import { Switch } from './switch'
import { Tab } from './tab'
import { Tabs } from './tabs'
import { Tag } from './tag'
import { Teleport } from './teleport'
import { ThemeProvide } from './theme-provide'
import { Toast } from './toast'
import { Tooltip } from './tooltip'
import { User } from './user'
const components = [
  Avatar,
  AvatarGroup,
  BackTop,
  Badge,
  BadgeAnchor,
  Breadcrumbs,
  BreadcrumbsItem,
  Button,
  Capacity,
  Card,
  Checkbox,
  CheckboxGroup,
  Code,
  Col,
  Collapse,
  CollapseGroup,
  Dot,
  Drawer,
  Form,
  FormItem,
  Grid,
  GridGroup,
  Image,
  ImageBrowser,
  Input,
  Link,
  Loading,
  Modal,
  Pagination,
  Popover,
  Progress,
  Radio,
  RadioGroup,
  Rating,
  Row,
  Select,
  SelectOption,
  Skeleton,
  SkeletonItem,
  Slider,
  Snippet,
  Spacer,
  Spinner,
  Swipe,
  SwipeItem,
  Switch,
  Tab,
  Tabs,
  Tag,
  Teleport,
  ThemeProvide,
  Toast,
  Tooltip,
  User
]

const install = (app: App) => {
  components.map((component: any) => {
    if (component.install) {
      app.use(component)
    } else if (component.name) {
      app.component(component.name, component)
    }
  })
}

export * from './avatar'
export * from './avatar-group'
export * from './back-top'
export * from './badge'
export * from './badge-anchor'
export * from './breadcrumbs'
export * from './breadcrumbs-item'
export * from './button'
export * from './capacity'
export * from './card'
export * from './checkbox'
export * from './checkbox-group'
export * from './code'
export * from './col'
export * from './collapse'
export * from './collapse-group'
export * from './dot'
export * from './drawer'
export * from './form'
export * from './form-item'
export * from './grid'
export * from './grid-group'
export * from './image'
export * from './image-browser'
export * from './input'
export * from './link'
export * from './loading'
export * from './modal'
export * from './pagination'
export * from './popover'
export * from './progress'
export * from './radio'
export * from './radio-group'
export * from './rating'
export * from './row'
export * from './select'
export * from './select-option'
export * from './skeleton'
export * from './skeleton-item'
export * from './slider'
export * from './snippet'
export * from './spacer'
export * from './spinner'
export * from './swipe'
export * from './swipe-item'
export * from './switch'
export * from './tab'
export * from './tabs'
export * from './tag'
export * from './teleport'
export * from './theme-provide'
export * from './toast'
export * from './tooltip'
export * from './user'
export * from './utils/composables'

export { install, version }

export default {
  install,
  version
}
