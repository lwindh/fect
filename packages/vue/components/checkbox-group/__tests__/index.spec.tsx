import { mount } from '@vue/test-utils'
import { ref } from 'vue'
import CheckboxGroup from '../index'
import Checkbox from '../../checkbox'
describe('CheckboxGroup', () => {
  it('should be render as element', () => {
    const el = mount(CheckboxGroup)
    expect(() => el.unmount()).not.toThrow()
  })

  it('should be support useRow', () => {
    const el = mount(CheckboxGroup, {
      props: {
        useRow: true
      }
    })
    expect(el.find('.useRow').exists()).toBe(true)
    expect(el.html()).toMatchSnapshot()
  })
  it('should support global size control', () => {
    const wrapper = mount({
      render() {
        return (
          <>
            <CheckboxGroup size={'mini'}>
              <Checkbox />
            </CheckboxGroup>
          </>
        )
      }
    })
    const el = wrapper.find('.fect-checkbox')
    expect(el.attributes('style')).toBe('--checkboxSize: 12px;')
  })

  it('should use v-model to control all checkbox', async () => {
    const wrapper = mount({
      setup() {
        const parent = ref(['xiamen', 'shanghai'])
        return { parent }
      },
      render() {
        return (
          <CheckboxGroup v-model={this.parent}>
            {this.parent.map((p, idx) => (
              <Checkbox label={p}>{idx}</Checkbox>
            ))}
          </CheckboxGroup>
        )
      }
    })
    const els = wrapper.findAll('[type="checkbox"]')
    expect(wrapper.vm.parent).toEqual(['xiamen', 'shanghai'])
    await els[0].trigger('change')
    expect(wrapper.vm.parent).toEqual(['shanghai'])
    await els[1].trigger('change')
    expect(wrapper.vm.parent).toEqual([])
    await els[0].trigger('change')
    expect(wrapper.vm.parent).toEqual(['shanghai'])
  })
})
