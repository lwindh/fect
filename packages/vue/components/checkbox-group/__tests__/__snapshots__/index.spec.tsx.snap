// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup should be support useRow 1`] = `
<div class="fect-checkbox__group useRow">
  <!---->
</div>
`;
